src/main/groovy/com/wikia/gradle/marathon/MarathonTask.groovy

```diff
@@ -47,6 +49,12 @@ class MarathonTask extends DefaultTask {
             app.setHealthChecks(healthChecks)
         }
 
+        List<List<String>> constraints = this.stage.resolve(Constraints).getConstraints()
```

Comment:
If app doesn't specify constraints it will not throw exception. I did however found that due to a bug we could not clear them. Fixed in the following commit.